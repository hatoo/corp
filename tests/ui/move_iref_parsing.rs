use futures::FutureExt;
use stap::{Cursor, Input, Parsing};

fn main() {
    let mut input = Input::new(Cursor {
        buf: Vec::<u8>::new(),
        index: 0,
    });

    let mut p = Parsing::new(&mut input, |iref| async move { iref }.boxed_local());

    p.poll();

    let iref = p.into_result().unwrap();
    drop(input);
    iref;
}
