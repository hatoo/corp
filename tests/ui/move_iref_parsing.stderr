error[E0505]: cannot move out of `input` because it is borrowed
  --> tests/ui/move_iref_parsing.rs:15:10
   |
5  |     let mut input = Input::new(Cursor {
   |         --------- binding `input` declared here
...
10 |     let mut p = Parsing::new(&mut input, |iref| async move { iref }.boxed_local());
   |                              ---------- borrow of `input` occurs here
...
15 |     drop(input);
   |          ^^^^^ move out of `input` occurs here
16 |     iref;
   |     ---- borrow later used here
